tests/solvers/linear/univariate_test.py

```diff
@@ -163,3 +168,31 @@ def univ_dist(x, y, a, b):
     expected = univ_dist(x, y, a, b + v_b) - univ_dist(x, y, a, b - v_b)
     actual = 2.0 * jnp.vdot(v_b, grad_b)
     np.testing.assert_allclose(actual, expected, rtol=tol, atol=tol)
+
+  @pytest.mark.fast()
+  def test_dual_vectors(self):
+    """The OTT solver outputs meaningful dual variables."""
+    _ = pytest.importorskip("lineax")  # only tested using lineax
+    n = self.n
+    x, a, y, b, z, c = self.x, self.a, self.y, self.b, self.z, self.c
+    unif_n = jnp.ones((n,)) / n
+    for (target, weights_target) in ((y, b), (z, c)):
+      for weights_source in (a, unif_n):
+        geom = pointcloud.PointCloud(x, target, cost_fn=costs.SqEuclidean())
+        prob = linear_problem.LinearProblem(
+            geom=geom, a=weights_source, b=weights_target
+        )
+        out = univariate.UnivariateSolver()(prob, return_dual_vectors=True)
+
+        f, g = out.dual_a, out.dual_b
+        dual_obj = jnp.sum(f * weights_source[None, :], axis=1)
```

Comment:
Might be beneficial to add this as a property of the `UnivariateOutput` instead.

Reply:
this is just for testing, in principle this `dual_obj` should match exactly `ot_costs` (computed differently), so no need to create a property